SignalMessaging/attachments/MessageApprovalViewController.swift

```diff
@@ -0,0 +1,231 @@
+        recipientRow.addSubview(topBorder)
+        topBorder.autoPinWidthToSuperview()
+        topBorder.autoPinTopToSuperview()
+        topBorder.autoSetDimension(.height, toSize: borderThickness)
+
+        let bottomBorder = UIView.container()
+        bottomBorder.backgroundColor = borderColor
+        recipientRow.addSubview(bottomBorder)
+        bottomBorder.autoPinWidthToSuperview()
+        bottomBorder.autoPinBottomToSuperview()
+        bottomBorder.autoSetDimension(.height, toSize: borderThickness)
+
+        guard let font = UIFont.ows_regularFont(withSize:ScaleFromIPhone5To7Plus(14.0, 18.0)) else {
+            owsFail("Can't load font")
+            return recipientRow
+        }
+        let hSpacing = CGFloat(10)
+        let hMargin = CGFloat(15)
+        let vSpacing = CGFloat(5)
+        let vMargin = CGFloat(10)
+
+        let toLabel = UILabel()
+        toLabel.text = NSLocalizedString("MESSAGE_APPROVAL_RECIPIENT_LABEL",
+                                         comment: "Label for the recipient name in the 'message approval' dialog.")
+        toLabel.textColor = UIColor.ows_darkGray
+        toLabel.font = font
+        recipientRow.addSubview(toLabel)
+
+        let nameLabel = UILabel()
+        nameLabel.textColor = UIColor.black
+        nameLabel.font = font
+        nameLabel.lineBreakMode = .byTruncatingTail
+        recipientRow.addSubview(nameLabel)
+
+        toLabel.autoPinLeadingToSuperview(withMargin: hMargin)
+        toLabel.setContentHuggingHorizontalHigh()
+        toLabel.setCompressionResistanceHorizontalHigh()
+        toLabel.autoAlignAxis(.horizontal, toSameAxisOf: nameLabel)
+
+        nameLabel.autoPinLeading(toTrailingOf: toLabel, margin:hSpacing)
+        nameLabel.autoPinTrailingToSuperview(withMargin: hMargin)
+        nameLabel.setContentHuggingHorizontalLow()
+        nameLabel.setCompressionResistanceHorizontalLow()
+        nameLabel.autoPinTopToSuperview(withMargin: vMargin)
+
+        if let groupThread = self.thread as? TSGroupThread {
+            let groupName = (groupThread.name().count > 0
+            ? groupThread.name()
+                : MessageStrings.newGroupDefaultTitle)
+
+            nameLabel.text = groupName
+            nameLabel.autoPinBottomToSuperview(withMargin: vMargin)
+
+            return recipientRow
+        }
+        guard let contactThread = self.thread as? TSContactThread else {
+            owsFail("Unexpected thread type")
+            return recipientRow
+        }
+
+        nameLabel.attributedText = contactsManager.formattedFullName(forRecipientId:contactThread.contactIdentifier(), font:font)
+        nameLabel.textColor = UIColor.black
+
+        if let profileName = self.profileName(contactThread:contactThread) {
+            // If there's a profile name worth showing, add it as a second line below the name.
+            let profileNameLabel = UILabel()
+            profileNameLabel.textColor = UIColor.ows_darkGray
```

Comment:
It's not a pixel-for-pixel match, but I've followed the pattern that we used in `ContactTableViewCell` of showing the optional "profile name" below the "name", with a lighter color.

Reply:
We're eventually going to allow multiple recipients, right?

In which case we'll potentially need to wrap the list of names to multiple lines. I think in that context having the profile name below the phone number will be strange, and it would be preferable for the profile name to be on the same line as the phone number:

ASCII mockup:

e.g. wrapping a long line of recipients:

```
To: +1-555-555-5555 ~Alice, Bob,
Charles, Danielle 
```

A little awkward when only some recipients have profile names.

```
To: +1-555-555-5555, Bob,
    ~Alice
Charles, Danielle 
```

I guess I'm envisioning it would look like a `To:` line in any other messaging/email app.

I'm ok with the current approach if you'd rather not do that work now.

Reply:
Great points.  I hesitated to combine them onto a single line for a few reasons.  One is to feel consistent with how these names are presented in the main app (currently).  Another is because I seem to recall security concerns around ambiguity between where one ended and another begun (which could be mitigated by font, font weight, font size, color, etc.).

Let's revisit this when we do the work to allow multiple recipients.